src/codeflare_sdk/cli/cli_utils.py

```diff
@@ -73,3 +79,95 @@ def resolve_command(self, ctx, args):
         # always return the full command name
         _, cmd, args = super().resolve_command(ctx, args)
         return cmd.name, cmd, args
+
+
+def print_jobs(jobs):
+    headers = ["Submission ID", "Job ID", "RayCluster", "Namespace", "Status"]
+    table = Table(show_header=True)
+    for header in headers:
+        table.add_column(header)
+    for job in jobs:
+        table.add_row(*[job[header] for header in headers])
+    print(table)
+
+
+def list_all_kubernetes_jobs(print_to_console=True):
+    k8s_jobs = []
+    runner = get_runner()
+    jobs = runner.list(scheduler="kubernetes_mcad")
+    rayclusters = {
+        raycluster.name for raycluster in list_clusters_all_namespaces(False)
+    }
+    for job in jobs:
+        namespace, name = job.app_id.split(":")
+        status = job.state
+        if name not in rayclusters:
+            k8s_jobs.append(
+                {
+                    "Submission ID": name,
+                    "Job ID": "N/A",
+                    "RayCluster": "N/A",
+                    "Namespace": namespace,
+                    "Status": str(status),
+                    "App Handle": job.app_handle,
+                }
+            )
+    if print_to_console:
+        print_jobs(k8s_jobs)
+    return k8s_jobs
+
+
+def list_all_jobs(print_to_console=True):
+    k8s_jobs = list_all_kubernetes_jobs(False)
+    rc_jobs = list_all_raycluster_jobs(False)
+    all_jobs = rc_jobs + k8s_jobs
+    if print_to_console:
+        print_jobs(all_jobs)
+    return all_jobs
+
+
+def list_raycluster_jobs(cluster: RayCluster, print_to_console=True):
+    rc_jobs = []
+    client = JobSubmissionClient(cluster.dashboard)
+    jobs = client.list_jobs()
+    for job in jobs:
+        job_obj = {
+            "Submission ID": job.submission_id,
+            "Job ID": job.job_id,
+            "RayCluster": cluster.name,
+            "Namespace": cluster.namespace,
+            "Status": str(job.status),
+            "App Handle": "ray://torchx/" + cluster.dashboard + "-" + job.submission_id,
+        }
+        rc_jobs.append(job_obj)
+    if print_to_console:
+        print_jobs(rc_jobs)
+    return rc_jobs
```

Comment:
There is already a function for this in the cluster class `list_jobs()` can that be reused here instead?

Reply:
@MichaelClifford Cluster's `list_jobs()` function returns jobs in a format that doesn't work with my table and doesn't include things like app handle. So this function is mainly reformatting the jobs so they work with a table and getting extra data.

Reply:
Ok. We may want to look into some refactoring in the near future then, so that we are only maintaining a single function that gets_jobs info. But this is fine for now.